packages/core/addon/models/table.js

```diff
@@ -77,22 +78,63 @@ export default VisualizationBase.extend(Validations, {
   }
 });
 
+/**
+ * returns default dimension fields (show clause)
+ * @param {Object} dimension - dimension from request
+ * @returns {Object} - list of field names
+ */
+function getDefaultDimensionFields(dimension) {
+  let fields = get(dimension, 'dimension.fields');
+
+  if (isArray(fields)) {
+    return fields.filter(field => isArray(field.tags) && field.tags.includes('show')).map(field => field.name);
+  }
+
+  return [];
+}
+
+/**
+ * builds a single dimension column
+ * @param {Object} dimension - dimension from request
+ * @param {Object} column
+ * @param {String} fieldName
+ * @returns {Object} - dimension column
+ */
+function dimensionColumn(dimension, column, fieldName) {
+  let defaultName = dimensionFormat(get(dimension, 'dimension.longName'), fieldName);
+
+  return {
+    field: { dimension: get(dimension, 'dimension.name') },
+    type: 'dimension',
+    displayName: column ? column.displayName : defaultName
+  };
+}
+
 /**
  * builds dimension columns for new visualization builds
  * @param {Array} dimensions - dimensions from request
  * @param {Object} columnIndex - column lookup table indexed by dimension/metric name
  * @returns {Array} - list of dimensions columns
  */
 function buildDimensionColumns(dimensions, columnIndex) {
-  return dimensions.map(dimension => {
-    let column = columnIndex[get(dimension, 'dimension.name')];
-
-    return {
-      field: { dimension: get(dimension, 'dimension.name') },
-      type: 'dimension',
-      displayName: column ? column.displayName : get(dimension, 'dimension.longName')
-    };
+  let dimensionColumns = dimensions.map(dimension => {
+    let defaultFields = getDefaultDimensionFields(dimension),
+      dimensionName = get(dimension, 'dimension.name');
+
+    if (!defaultFields.length) {
+      let column = columnIndex[dimensionName];
+      return dimensionColumn(dimension, column);
+    } else {
+      //return an array of columns for each dimension field
+      return defaultFields.map(fieldName => {
+        let column = columnIndex[canonicalizeDimension(dimensionName, fieldName)];
+        return merge({}, { field: { show: fieldName } }, dimensionColumn(dimension, column, fieldName));
```

Comment:
should this logic be in `dimensionColumn`?

Reply:
sounds better